cloudsmith/data_source_package_download_test.go

```diff
@@ -0,0 +1,196 @@
+					resource.TestCheckResourceAttr("cloudsmith_repository.test", "namespace", namespace),
+				),
+			},
+			{
+				PreConfig: func() {
+					createHelloWorldFile(fileName)
+					uploadPackageToCloudsmith(apiKey, namespace, repoName, fileName, packageName, packageVersion, userName)
+				},
+				Config: testAccPackageDownloadData(namespace, repoName, packageName, packageVersion, destinationPath),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.cloudsmith_package_download.test", "namespace", namespace),
+					resource.TestCheckResourceAttr("data.cloudsmith_package_download.test", "repository", repoName),
+					resource.TestCheckResourceAttr("data.cloudsmith_package_download.test", "package_name", packageName),
+					resource.TestCheckResourceAttr("data.cloudsmith_package_download.test", "package_version", packageVersion),
+					resource.TestCheckResourceAttr("data.cloudsmith_package_download.test", "destination_path", destinationPath),
+					checkDownloadedFile(namespace, repoName, packageName, packageVersion, destinationPath, "Hello world"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourcePackageDownload_repositoryConfig(repoName string, namespace string) string {
+	return fmt.Sprintf(`
+resource "cloudsmith_repository" "test" {
+	name      = "%s"
+	namespace = "%s"
+}
+`, repoName, namespace)
+}
+
+func createHelloWorldFile(filename string) {
+	content := []byte("Hello world")
+	err := os.WriteFile(filename, content, 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func uploadPackageToCloudsmith(apiKey, namespace, repoName, fileName, packageName, packageVersion string, userName string) {
+	client := &http.Client{}
+
+	// Calculate the SHA-256 checksum of the file
+	fileData, err := os.ReadFile(fileName)
+	if err != nil {
+		panic(err)
+	}
+	sha256Checksum := fmt.Sprintf("%x", sha256.Sum256(fileData))
+
+	// Step 1: PUT (upload) the file
+	uploadURL := fmt.Sprintf("https://upload.cloudsmith.io/%s/%s/%s", namespace, repoName, fileName)
+	request, err := http.NewRequest("PUT", uploadURL, bytes.NewReader(fileData))
+	if err != nil {
+		panic(err)
+	}
+	request.SetBasicAuth(userName, apiKey)
+	request.Header.Set("Content-Type", "application/octet-stream")
+	request.Header.Set("Content-Sha256", sha256Checksum)
+
+	response, err := client.Do(request)
+	if err != nil {
+		panic(err)
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		panic(fmt.Errorf("failed to upload package: %s", response.Status))
+	}
+
+	// Read the response body
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		panic(err)
+	}
+
+	// Parse the response JSON
+	var uploadResponse map[string]string
+	err = json.Unmarshal(body, &uploadResponse)
+	if err != nil {
+		panic(err)
+	}
+	identifier := uploadResponse["identifier"]
+
+	// Step 2: POST the package details
+	createPackageURL := fmt.Sprintf("https://api-prd.cloudsmith.io/v1/packages/%s/%s/upload/raw/", namespace, repoName)
```

Comment:
can we avoid this hardcoded URL here? otherwise the provider won't work against stg/dev/local envs